Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -0,0 +1,246 @@
+    break;
+
+  case IOCTL_NWC24_UNLOCK_SOCKET:
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_UNLOCK_SOCKET - NI");
+    break;
+
+  case IOCTL_NWC24_REQUEST_REGISTER_USER_ID:
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_REGISTER_USER_ID");
+    Memory::Write_U32(0, request.buffer_out);
+    Memory::Write_U32(0, request.buffer_out + 4);
+    break;
+
+  case IOCTL_NWC24_REQUEST_GENERATED_USER_ID:  // (Input: none, Output: 32 bytes)
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_GENERATED_USER_ID");
+    if (config.CreationStage() == NWC24::NWC24Config::NWC24_IDCS_INITIAL)
+    {
+      const std::string settings_file_path(
+          Common::GetTitleDataPath(TITLEID_SYSMENU, Common::FROM_SESSION_ROOT) + WII_SETTING);
+      SettingsHandler gen;
+      std::string area, model;
+      bool got_settings = false;
+
+      if (File::Exists(settings_file_path) && gen.Open(settings_file_path))
+      {
+        area = gen.GetValue("AREA");
+        model = gen.GetValue("MODEL");
+        got_settings = true;
+      }
+      if (got_settings)
+      {
+        u8 area_code = GetAreaCode(area);
+        u8 id_ctr = config.IdGen();
+        u8 hardware_model = GetHardwareModel(model);
+
+        const EcWii& ec = EcWii::GetInstance();
+        u32 HollywoodID = ec.GetNGID();
+        u64 UserID = 0;
+
+        s32 ret = NWC24MakeUserID(&UserID, HollywoodID, id_ctr, hardware_model, area_code);
+        config.SetId(UserID);
+        config.IncrementIdGen();
+        config.SetCreationStage(NWC24::NWC24Config::NWC24_IDCS_GENERATED);
+        config.WriteConfig();
+
+        Memory::Write_U32(ret, request.buffer_out);
+      }
+      else
+      {
+        Memory::Write_U32(NWC24::WC24_ERR_FATAL, request.buffer_out);
+      }
+    }
+    else if (config.CreationStage() == NWC24::NWC24Config::NWC24_IDCS_GENERATED)
+    {
+      Memory::Write_U32(NWC24::WC24_ERR_ID_GENERATED, request.buffer_out);
+    }
+    else if (config.CreationStage() == NWC24::NWC24Config::NWC24_IDCS_REGISTERED)
+    {
+      Memory::Write_U32(NWC24::WC24_ERR_ID_REGISTERED, request.buffer_out);
+    }
+    Memory::Write_U64(config.Id(), request.buffer_out + 4);
+    Memory::Write_U32(config.CreationStage(), request.buffer_out + 0xC);
+    break;
+
+  case IOCTL_NWC24_GET_SCHEDULAR_STAT:
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_GET_SCHEDULAR_STAT - NI");
+    break;
+
+  case IOCTL_NWC24_SAVE_MAIL_NOW:
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_SAVE_MAIL_NOW - NI");
+    break;
+
+  case IOCTL_NWC24_REQUEST_SHUTDOWN:
+    // if ya set the IOS version to a very high value this happens ...
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN - NI");
+    break;
+
+  default:
+    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
+  }
+
+  return GetDefaultReply(return_value);
+}
+
+u8 NetKDRequest::GetAreaCode(const std::string& area) const
+{
+  static std::map<const std::string, u8> regions = {
```

Comment:
Sure, I can do that in a different PR. This PR is just intended to move code with no differing functional changes.